source/mainnet/smart-contracts/tutorials/piggy-bank/preparing.rst

```diff
@@ -0,0 +1,231 @@
+.. _Rust: https://www.rust-lang.org/
+
+.. _piggy-bank-preparing:
+
+====================================
+Preparing a testnet node on a server
+====================================
+
+This is the third :ref:`part of a tutorial<piggy-bank>` on smart contract
+development.
+So far you have written and tested a piggy bank smart contract in the Rust_ programming
+language.
+This part will focus on how you can set up your own testnet node on a server, and create a testnet account.
+In more detail, you will create an instance on a server that will host your own testnet node. Secondly, you will create your own testnet account and import the keys to your testnet node.
+
+.. warning::
+
+   The reader is assumed to have basic knowledge of what a blockchain and smart
+   contract is, and some experience with Rust_.
+
+
+Preparation
+===========
+
+Before you start, make sure you create an account at your favourite cloud provider to set up your instance.
+
+You are now ready to select an instance on the cloud provider that matches the following requirements to run a testnet node:
+
+.. list-table::
+   :widths: 25 25
+   :header-rows: 1
+
+   * - Hardware (Testnet node)
+     - Recommended
+   * - CPU (Core)
+     - ?
+   * - RAM (Memory)
+     - 8 GB
+   * - Storage
+     - 50 GB
+   * - Operation system
+     - Ubuntu 20.04 x64
+
+.. Note::
+   The above requirements are simple and meant for a testnet node only. Please have a look at the requirements for mainnet nodes in the :ref:`run a node <node-requirements>` chapter.
+
+Syncing a testnet node
+======================
+
+The :ref:`run a node <node-requirements>` guide will help you set up your instance correctly. It is recommended using an ubuntu instance on a server in the cloud for this tutorial.
+
+.. Note::
+   It is technically fine to run your testnet node locally on your machine instead of on a server in the cloud. Since blockchain nodes have to run 24/7 to be up-to-date with the blockchain, you have to run your local machine 24/7. Alternatively if you don't want to run your local machine 24/7, you can let catch up your node whenever you start your machine. Because this takes some time, this tutorial recommends a cloud provider setup for convenience.
+
+Start the syncing process of the testnet node by following the guide for your platform :ref:`Ubuntu<ubuntu-node>`, :ref:`Docker<docker-node>`, :ref:`Windows<windows-node>`, or :ref:`MacOS<macos-node>`.
+
+You should find your node name on the `Concordium testnet dashboard <https://dashboard.testnet.concordium.com/>`__. It will take around 1-2 days until your testnet node is fully synced. You can observe the syncing process by watching the finalization length of your node. Wait until the `Fin Length` (finalization length) of your node is the same as the highest value used by the majority of nodes.
```

Comment:
This is only true if they are running the collector which they don't have to, and they can disable.

Reply:
You can add that if they aren't running the collector service (which is the one sending data from the node to the dashboard), then they can query the block height directly from the node and compare _that_ value to the one on the dashboard.

```
$ concordium-client block show
...
Height:                     24924
...
```